<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from ur5_soma.urdf.xacro            | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="ur5_soma" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <!-- include other model components -->
  <material name="SoftHand/LightGrey">
    <color rgba="0.7 0.7 0.7 1.0"/>
  </material>
  <material name="SoftHand/DarkGrey">
    <color rgba="0.4 0.4 0.4 1.0"/>
  </material>
  <material name="SoftHand/Black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="SoftHand/DarkGolden">
    <color rgba="0.4 0.4 0.3 1.0"/>
  </material>
  <material name="SoftHand/Blue">
    <color rgba="0.0 0.0 0.8 0.9"/>
  </material>
  <material name="SoftHand/LightBlue">
    <color rgba="0.0 0.4 1.0 0.5"/>
  </material>
  <!-- configure how all joints will behave in gazebo (particular behavior can be set at each joint as well) -->
  <!-- NOTE: inertia values are approximated using meshlab and only using the significant elements -->
  <!-- from http://answers.ros.org/question/11350/is-there-any-way-to-calculate-inertial-property-of-a-robot-to-simulate-it-in-gazebo/s
    One general rule of thumb I use for checking my inertia tensors is: If total mass of the rigid body is m, and the dimension of the corresponding body is d, then check to see if ixx, iyy, izz are near m*(d/2)^2.

    This is by no means correct, but a sanity check to make sure the moment of inertia are the right order of magnitudes, so the model behaves somewhat physically realistically.
    -->
  <!-- conversions -->
  <!-- I couldn't make the simulation work with realistic values, so I tweaked the conversion values -->
  <!-- not so negligible, otherwise simulation fails -->
  <!-- MESHLAB RESULT 
Opened mesh /home/crosales/Code/catkin_ws/src/pisa-iit-soft-hand/soft_hand/soft_hand_description/meshes/knuckle.stl in 9 msec
All files opened in 1398 msec
Enabled Decorate mode Show Axis
Mesh Bounding Box Size 34.382004 14.000006 24.628571
Mesh Bounding Box Diag 44.549847 
Mesh Volume is 3450.446777
Mesh Surface is 4583.448730
Thin shell barycenter 1.201245 0.001308 0.586193
Center of Mass is 1.285644 -0.000958 0.425011
Inertia Tensor is :
| 160373.578125 -18.966606 13918.184570 |
| -18.966606 311858.187500 59.538574 |
| 13918.184570 59.538574 249686.000000 |
Principal axes are :
| 0.988611 -0.000038 0.150492 |
| 0.000180 1.000000 -0.000933 |
| -0.150492 0.000949 0.988611 |
axis momenta are :
| 158254.859375 311858.250000 251804.656250 |
-->
  <!-- MESHLAB RESULT 
Opened mesh /home/crosales/Code/catkin_ws/src/pisa-iit-soft-hand/soft_hand/soft_hand_description/meshes/thumb_knuckle.stl in 6 msec
All files opened in 1493 msec
Enabled Decorate mode Show Axis
Mesh Bounding Box Size 38.298431 21.449932 21.967697
Mesh Bounding Box Diag 49.086140 
Mesh Volume is 3444.315186
Mesh Surface is 4292.456543
Thin shell barycenter 14.662230 -0.440256 0.060471
Center of Mass is 13.608098 -0.532174 -0.216691
Inertia Tensor is :
| 121779.453125 -25104.320312 -7351.798340 |
| -25104.320312 336414.968750 4401.581055 |
| -7351.798340 4401.581055 333794.062500 |
Principal axes are :
| 0.992996 -0.113512 0.032781 |
| 0.113844 0.845023 -0.522470 |
| 0.031606 0.522542 0.852027 |
axis momenta are :
| 118667.312500 342509.062500 330812.125000 |
-->
  <!-- MESHLAB RESULT 
Opened mesh /home/crosales/Code/catkin_ws/src/pisa-iit-soft-hand/soft_hand/soft_hand_description/meshes/phalanx.stl in 8 msec
All files opened in 1240 msec
Mesh Bounding Box Size 25.158094 14.000006 18.515181
Mesh Bounding Box Diag 34.230713 
Mesh Volume is 2003.388916
Mesh Surface is 2495.472412
Thin shell barycenter 5.001481 0.001261 0.002639
Center of Mass is 5.074337 -0.004524 -0.262953
Inertia Tensor is :
| 79493.273438 -2.796152 -5974.423340 |
| -2.796152 121398.460938 16.481358 |
| -5974.423340 16.481358 122965.343750 |
Principal axes are :
| 0.991019 0.000928 -0.133714 |
| 0.000013 0.999975 0.007040 |
| 0.133718 -0.006979 0.990995 |
axis momenta are :
| 78687.148438 121398.343750 123771.585938 |
-->
  <!-- MESHLAB RESULT 
Opened mesh /home/crosales/Code/catkin_ws/src/pisa-iit-soft-hand/soft_hand/soft_hand_description/meshes/fingertip.stl in 7 msec
All files opened in 1241 msec
Mesh Bounding Box Size 32.299561 14.000006 19.228573
Mesh Bounding Box Diag 40.112339 
Mesh Volume is 3519.114502
Mesh Surface is 4175.043945
Thin shell barycenter 8.126755 0.000146 0.955067
Center of Mass is 9.849094 0.000560 0.820286
Inertia Tensor is :
| 133971.296875 -13.962580 -14369.020508 |
| -13.962580 308096.281250 -12.146572 |
| -14369.020508 -12.146572 287758.875000 |
Principal axes are :
| 0.995737 -0.000033 -0.092238 |
| 0.000086 1.000000 0.000569 |
| 0.092238 -0.000574 0.995737 |
axis momenta are :
| 132640.250000 308096.281250 289089.906250 |
-->
  <!-- MESHLAB RESULT 
Opened mesh /home/crosales/Code/catkin_ws/src/pisa-iit-soft-hand/soft_hand/soft_hand_description/meshes/palm_right.stl in 13 msec
All files opened in 1178 msec
Mesh Bounding Box Size 36.539444 91.129791 84.829025
Mesh Bounding Box Diag 129.752579 
Mesh Volume is 76748.765625
Mesh Surface is 52111.375000
Thin shell barycenter -5.584415 -7.926011 28.832655
Center of Mass is -4.568758 -6.808116 27.653328
Inertia Tensor is :
| 55001040.000000 -714369.625000 -867280.437500 |
| -714369.625000 31722332.000000 -1570798.250000 |
| -867280.437500 -1570798.250000 37379544.000000 |
Principal axes are :
| 0.998540 0.038402 0.037998 |
| -0.027432 0.966351 -0.255759 |
| -0.046541 0.254343 0.965993 |
axis momenta are :
| 55061088.000000 31280510.000000 37761316.000000 |
-->
  <!-- accesories -->
  <!-- joint properties -->
  <!-- synergy vector -->
  <!-- synergy joint properties -->
  <!-- SOFT FINGER model -->
  <!-- 4 fingers are equal (index, middle, ring, little) -->
  <!-- THUMB MODEL -->
  <!-- It has one joint less than rest of fingers -->
  <!-- SOFT HAND MODEL -->
  <!-- Include all available models -->
  <!-- Common tools: PI, inertias of a cuboid and cylinder -->
  <!-- ############# -->
  <!-- Sensor models -->
  <!-- ############# -->
  <!-- Nano17: http://www.ati-ia.com/products/ft/ft_models.aspx?id=Nano17 -->
  <!-- Base model for all sensor to have common frame names -->
  <!-- ur5 -->
  <!-- Inertia parameters -->
  <!-- This mass might be incorrect -->
  <!-- Kinematic model -->
  <!-- Properties from urcontrol.conf -->
  <!--
    DH for UR5:
    a = [0.00000, -0.42500, -0.39225,  0.00000,  0.00000,  0.0000]
    d = [0.089159,  0.00000,  0.00000,  0.10915,  0.09465,  0.0823]
    alpha = [ 1.570796327, 0, 0, 1.570796327, -1.570796327, 0 ]
    q_home_offset = [0, -1.570796327, 0, -1.570796327, 0, 0]
    joint_direction = [-1, -1, 1, 1, 1, 1]
    mass = [3.7000, 8.3930, 2.2750, 1.2190, 1.2190, 0.1879]
    center_of_mass = [ [0, -0.02561, 0.00193], [0.2125, 0, 0.11336], [0.11993, 0.0, 0.0265], [0, -0.0018, 0.01634], [0, 0.0018,0.01634], [0, 0, -0.001159] ]
  -->
  <!-- Arbitrary offsets for shoulder/elbow joints -->
  <!-- measured from model -->
  <!-- measured from model -->
  <!-- link lengths used in model -->
  <!--property name="shoulder_height" value="0.089159" /-->
  <!--property name="shoulder_offset" value="0.13585" /-->
  <!-- shoulder_offset - elbow_offset + wrist_1_length = 0.10915 -->
  <!--property name="upper_arm_length" value="0.42500" /-->
  <!--property name="elbow_offset" value="0.1197" /-->
  <!-- CAD measured -->
  <!--property name="forearm_length" value="0.39225" /-->
  <!--property name="wrist_1_length" value="0.093" /-->
  <!-- CAD measured -->
  <!--property name="wrist_2_length" value="0.09465" /-->
  <!-- In CAD this distance is 0.930, but in the spec it is 0.09465 -->
  <!--property name="wrist_3_length" value="0.0823" /-->
  <!-- manually measured -->
  <!-- manually measured -->
  <!-- manually measured -->
  <!-- manually measured -->
  <!-- manually measured -->
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="ros_control">
      <!--robotNamespace>/</robotNamespace-->
      <!--robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType-->
    </plugin>
    <!--
    <plugin name="gazebo_ros_power_monitor_controller" filename="libgazebo_ros_power_monitor.so">
      <alwaysOn>true</alwaysOn>
      <updateRate>1.0</updateRate>
      <timeout>5</timeout>
      <powerStateTopic>power_state</powerStateTopic>
      <powerStateRate>10.0</powerStateRate>
      <fullChargeCapacity>87.78</fullChargeCapacity>     
      <dischargeRate>-474</dischargeRate>
      <chargeRate>525</chargeRate>
      <dischargeVoltage>15.52</dischargeVoltage>
      <chargeVoltage>16.41</chargeVoltage>
    </plugin>
-->
  </gazebo>
  <!-- arm -->
  <link name="base_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur5/visual/base.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur5/collision/base.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="4.0"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.00443333156" ixy="0.0" ixz="0.0" iyy="0.00443333156" iyz="0.0" izz="0.0072"/>
    </inertial>
  </link>
  <joint name="shoulder_pan_joint" type="revolute">
    <parent link="base_link"/>
    <child link="shoulder_link"/>
    <origin rpy="0.0 0.0 3.14159265" xyz="0.0 0.0 0.089159"/>
    <axis xyz="0 0 1"/>
    <limit effort="150.0" lower="-3.14159265" upper="3.14159265" velocity="3.15"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="shoulder_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur5/visual/shoulder.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur5/collision/shoulder.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3.7"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.010267495893" ixy="0.0" ixz="0.0" iyy="0.010267495893" iyz="0.0" izz="0.00666"/>
    </inertial>
  </link>
  <joint name="shoulder_lift_joint" type="revolute">
    <parent link="shoulder_link"/>
    <child link="upper_arm_link"/>
    <origin rpy="0.0 1.570796325 0.0" xyz="0.0 0.13585 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-3.14159265" upper="3.14159265" velocity="3.15"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="upper_arm_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur5/visual/upperarm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur5/collision/upperarm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="8.393"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.28"/>
      <inertia ixx="0.22689067591" ixy="0.0" ixz="0.0" iyy="0.22689067591" iyz="0.0" izz="0.0151074"/>
    </inertial>
  </link>
  <joint name="elbow_joint" type="revolute">
    <parent link="upper_arm_link"/>
    <child link="forearm_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.1197 0.425"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-3.14159265" upper="3.14159265" velocity="3.15"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="forearm_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur5/visual/forearm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur5/collision/forearm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="2.275"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.25"/>
      <inertia ixx="0.049443313556" ixy="0.0" ixz="0.0" iyy="0.049443313556" iyz="0.0" izz="0.004095"/>
    </inertial>
  </link>
  <joint name="wrist_1_joint" type="revolute">
    <parent link="forearm_link"/>
    <child link="wrist_1_link"/>
    <origin rpy="0.0 1.570796325 0.0" xyz="0.0 0.0 0.39225"/>
    <axis xyz="0 1 0"/>
    <limit effort="28.0" lower="-3.14159265" upper="3.14159265" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_1_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur5/visual/wrist1.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur5/collision/wrist1.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.219"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.111172755531" ixy="0.0" ixz="0.0" iyy="0.111172755531" iyz="0.0" izz="0.21942"/>
    </inertial>
  </link>
  <joint name="wrist_2_joint" type="revolute">
    <parent link="wrist_1_link"/>
    <child link="wrist_2_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.093 0.0"/>
    <axis xyz="0 0 1"/>
    <limit effort="28.0" lower="-3.14159265" upper="3.14159265" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_2_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur5/visual/wrist2.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur5/collision/wrist2.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.219"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.111172755531" ixy="0.0" ixz="0.0" iyy="0.111172755531" iyz="0.0" izz="0.21942"/>
    </inertial>
  </link>
  <joint name="wrist_3_joint" type="revolute">
    <parent link="wrist_2_link"/>
    <child link="wrist_3_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.09465"/>
    <axis xyz="0 1 0"/>
    <limit effort="28.0" lower="-3.14159265" upper="3.14159265" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_3_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur5/visual/wrist3.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur5/collision/wrist3.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.1879"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0171364731454" ixy="0.0" ixz="0.0" iyy="0.0171364731454" iyz="0.0" izz="0.033822"/>
    </inertial>
  </link>
  <joint name="ee_fixed_joint" type="fixed">
    <parent link="wrist_3_link"/>
    <child link="ee_link"/>
    <origin rpy="0.0 0.0 1.570796325" xyz="0.0 0.0823 0.0"/>
  </joint>
  <link name="ee_link">
    <collision>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.01 0 0"/>
    </collision>
  </link>
  <transmission name="shoulder_pan_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_pan_joint">
      <hardwareInterface>VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_pan_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="shoulder_lift_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_lift_joint">
      <hardwareInterface>VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_lift_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="elbow_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="elbow_joint">
      <hardwareInterface>VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="elbow_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_1_joint">
      <hardwareInterface>VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_2_joint">
      <hardwareInterface>VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_3_joint">
      <hardwareInterface>VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- nothing to do here at the moment -->
  <!-- ROS base_link to UR 'Base' Coordinates transform -->
  <link name="base"/>
  <joint name="base_link-base_fixed_joint" type="fixed">
    <!-- NOTE: this rotation is only needed as long as base_link itself is
                 not corrected wrt the real robot (ie: rotated over 180
                 degrees)
      -->
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="base"/>
  </joint>
  <!-- Frame coincident with all-zeros TCP on UR controller -->
  <link name="tool0"/>
  <joint name="wrist_3_link-tool0_fixed_joint" type="fixed">
    <origin rpy="-1.570796325 0 0" xyz="0 0.0823 0"/>
    <parent link="wrist_3_link"/>
    <child link="tool0"/>
  </joint>
  <link name="world"/>
  <joint name="world_joint" type="fixed">
    <parent link="world"/>
    <child link="base_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
  </joint>
  <joint name="wrist_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.0165"/>
    <parent link="tool0"/>
    <child link="ati_gamma_base"/>
    <limit effort="100" lower="-0.00001" upper="0.00001" velocity="100"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="ati_gamma_base">
    <inertial>
      <mass value="0.255"/>
      <origin xyz="0 0 0.0165"/>
      <inertia ixx="0.0001137484875" ixy="0" ixz="0" iyy="0.000181214475" iyz="0" izz="0.0001137484875"/>
    </inertial>
    <visual>
      <!-- origin xyz="0 0 0.00725" rpy="0 0 0"/ -->
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.033" radius="0.0377"/>
        <!--mesh filename="package://force_torque_sensor/model/meshes/ati_gamma.stl"/-->
      </geometry>
      <material name="blue">
        <color rgba="0 0 .8 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.0165"/>
      <geometry>
        <cylinder length="0.033" radius="0.0377"/>
      </geometry>
    </collision>
  </link>
  <joint name="wrist_sensor_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.0165"/>
    <parent link="ati_gamma_base"/>
    <child link="ati_gamma_measure"/>
    <limit effort="100" lower="-0.00001" upper="0.00001" velocity="100"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="ati_gamma_measure">
    <inertial>
      <mass value="0.255"/>
      <origin xyz="0 0 0.0165"/>
      <inertia ixx="0.0001137484875" ixy="0" ixz="0" iyy="0.000181214475" iyz="0" izz="0.0001137484875"/>
    </inertial>
    <visual>
      <!-- origin xyz="0 0 0.00725" rpy="0 0 0"/ -->
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.033" radius="0.0377"/>
        <!--mesh filename="package://force_torque_sensor/model/meshes/ati_gamma.stl"/-->
      </geometry>
      <material name="red">
        <color rgba="0.5 0 0.0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.0165"/>
      <geometry>
        <cylinder length="0.033" radius="0.0377"/>
      </geometry>
    </collision>
  </link>
  <!--xacro:ft_sensor parent="tool0" name="wrist" type="ati_gamma">
    <origin xyz="0 0 0.0165" rpy="0 0 0"/>    
  </xacro:ft_sensor-->
  <!--joint name="hand_attachment_joint" type="revolute">
    <parent link="wrist_ati_gamma_measure" />
    <child link = "hand_attachment" />
    <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    <limit lower="-0.00001" upper="0.00001" effort="100" velocity="100"/>
    <axis xyz="0 0 1" />
  </joint-->
  <!--xacro:soft_hand parent="wrist_ati_gamma_measure" name="soft_hand" left="true" withAdaptiveTransmission="false" useMimicTag="false">
    <origin xyz="0 0 0.0" rpy="0 0 0"/>
  </xacro:soft_hand-->
  <!-- Frame coincident with all-zeros TCP on UR controller -->
  <!--link name="hand_attachment">      

      <collision>
        <geometry>
          <box size="0.01 0.01 0.01"/>
        </geometry>
        <origin rpy="0 0 0" xyz="0.01 0 0"/>
      </collision>
    </link-->
</robot>

